### PR TITLE
[python] Implement im delegate for chip-device-ctrl

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
We don't have a mechanism for python script be able to acknowledge the command it send.

 #### Summary of Changes

Implemented interaction model delegate for python device controller, this should be available for repl as well.

Currently, it implemented the following three API:

```
ClearCommandStatus(commandHandle)
WaitCommandStatus(commandHandle)
WaitCommandIndexStatus(commandHandle, commandIndex)
```

In the future, these three apis can be embedded into device class in python.

This PR is waiting for #5767, please revierw second commit only.

 Fixes #<<<<<FILL ME IN  - issue number(s). If no issue, please create one>>>>>

### CODE REVIEW

File: src/controller/python/ChipDeviceController-InteractionModelDelegate.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    CHIP_ERROR CommandResponseProtocolError(const app::CommandSender * apCommandSender, uint8_t aCommandIndex) override;
+
+    CHIP_ERROR CommandResponseError(const app::CommandSender * apCommandSender, CHIP_ERROR aError) override;
+
+    CHIP_ERROR CommandResponseProcessed(const app::CommandSender * apCommandSender) override;
+
+    static PythonInteractionModelDelegate & Instance();
+
+private:
+    friend void pychip_InteractionModelDelegate_SetCommandResponseStatusCallback(
```
Comment (andy31415) [inline src/controller/python/ChipDeviceController-InteractionModelDelegate.h:47]: please avoid using 'friend' - it shows tight coupling.

you can make the Set*Callback methods public.

File: src/controller/python/BUILD.gn [hunk 2]
```diff
@@ -104,6 +106,7 @@ pw_python_action("python") {
         "chip/clusters/__init__.py",
         "chip/configuration/__init__.py",
         "chip/exceptions/__init__.py",
+        "chip/interaction_model/IMDelegate.py",
```
Comment (andy31415) [inline src/controller/python/BUILD.gn:109]: interaction_model/IMDelegate stutters (twice the IM). Also it looks like naming convention for modules/python sources is lowercase.

Use `interaction_model/delegate.py`

File: src/controller/python/ChipDeviceController-InteractionModelDelegate.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#include <app/InteractionModelDelegate.h>
+
+namespace chip {
+namespace Controller {
+
+struct __attribute__((packed)) CommandStatus
```
Comment (andy31415) [inline src/controller/python/ChipDeviceController-InteractionModelDelegate.h:6]: Consider adding a comment that packed is used to make python binary data generation easy (not worry about alignment)

File: src/controller/python/BUILD.gn (hunk 2 above)
Comment (erjiaqing) [reply src/controller/python/BUILD.gn]: Done, also moved ChipDeviceController-IMDelegate to this directory.

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**